data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -128,7 +128,10 @@ private boolean isEventNameCreated(final S3EventNotification.S3EventNotification
     }
 
     private S3ObjectReference populateS3Reference(final S3EventNotification.S3EventNotificationRecord s3EventNotificationRecord) {
-        return S3ObjectReference.fromBucketAndKey(s3EventNotificationRecord.getS3().getBucket().getName(),
-                s3EventNotificationRecord.getS3().getObject().getKey());
+        final S3EventNotification.S3Entity s3Entity = s3EventNotificationRecord.getS3();
+        return S3ObjectReference.bucketAndKey(s3Entity.getBucket().getName(),
+                s3Entity.getObject().getKey())
+                .owner(s3Entity.getBucket().getOwnerIdentity().getPrincipalId())
```

Comment:
I like the simplicity in this approach. I want to clarify the users experience. Data prepper will only support buckets from the same account as the SQS Queue. Data prepper will not support cross account. Is that correct?

Reply:
That is not correct.

The ownerIdentity can be a different account. But, whatever account it is will be verified when performing the GetObject request. It is verifying that the bucket is still owned by the account that originally added the object to S3.

Reply:
Are we planning to extend this later, to support other accounts to access the s3 buckets?

Reply:
Ahhh, perfect. I really like how simple this is. Nicely done @dlvenable!